scanpy/tests/test_clustering.py

```diff
@@ -20,6 +21,52 @@ def test_leiden_basic(adata_neighbors):
     sc.tl.leiden(adata_neighbors)
 
 
+def test_clustering_subset(adata_neighbors):
+    methods = (
+        (sc.tl.louvain, 'louvain'),
+        (sc.tl.leiden, 'leiden'),
+    )
+    for clustering, key in methods:
+        print('Clustering method ' + str(key))
+        clustering(adata_neighbors, key_added=key)
+        print(adata_neighbors)
+        print(adata_neighbors.obs[key].value_counts())
+
+        for c in np.unique(adata_neighbors.obs[key]):
+            print('Analyzing cluster ' + str(c))
```

Comment:
just do `print('Analyzing cluster', c)` please

Reply:
@flying-sheep, should there be any print statements in the tests? Admittedly there's one there now from me, but it was only left in by accident.

Reply:
@ivirshup I think they can be used since they get printed only in the full trace (IIRC). But probably pytest has better solutions?

Reply:
pytest is pretty good about printing out values from failed assertions. Maybe think about formalizing what you expect to see from a print statement into an assertion?